artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionFactoryImpl.java

```diff
@@ -771,7 +771,11 @@ private void reconnectSessions(final RemotingConnection oldConnection,
       ((CoreRemotingConnection) connection).syncIDGeneratorSequence(((CoreRemotingConnection) oldConnection).getIDGeneratorSequence());
 
       for (ClientSessionInternal session : sessionsToFailover) {
-         session.handleFailover(connection, cause);
+         if (session.handleFailover(connection, cause)) {
+            connection.destroy();
```

Comment:
Can you add a test for such condition?

Reply:
We'll add test to simulate the scenario where reattach timeout. Just not sure the proper way to fix the problem:)

Reply:
Well.. .the thing is... I"m not sure what was the problem to be able to tell you what is the fix. a test would help.

If you can't provide a test.. for something rare happening in production, it would be nice some more details.. like logs.. stack traces.... 


i couldn't figure out the issue by just the synchrnoized implementation.

Reply:
The synchrnoized implementation? Do you mean #2558 ?  This issue is easy to simulate:)

Reply:
I actually confused with this issue.. but yeah.. on both cases.. a test case... or more detailed information would help.


Think about this: another user is having a similar issue, if you only provide the fix and not what happened they will never be able to pinpoint this is the fix they need. for both cases.

Reply:
@clebertsuconic i think i have seen something similar in our production environment, unfortunatly because its prod they restarted the client app too quickly before details were gathered.

@wy96f as clebert notes, would be great to have a test case then i could also tell if it is what ive also seen.